### ISSUE TITLE
Enhancement Request: cli show/list content of pbxproj file

### ISSUE THREAD

**Issue by kwhat** [kronenthaler/mod-pbxproj#201]
Hi,

I was looking for a way to print out the content of a pbxproj file in a human readable format and came across this project.  It would be awesome if I could use this project to print out information about build targets like the src files required, flags, etc.  I don't believe this is currently supported by the cli program or the library.

Thanks!

**Comment by kronenthaler** [comment]
@kwhat I like the idea of display some info about the targets.
Something in the lines of:

```
$> pbxproj show
target_name_1:
    type: native target
    configurations: debug, release
    source files count: X
    header files count: Y
    library dependencies count: Z
    resource files count: R
...    
```

Maybe allow to dig into specific target by specifying the target name and what you want to list

```
$> pbxproj show target_name_1 (--source-files | --header-files | --libraries | --resources)
<file type> files: a, b, c, ... (or one per line)
```

As flags are specified per target-configuration pair, it would be necessary to pass the configuration name

```
$> pbxproj show target_name_1 --flags (Debug | Release | config_name)
<ConfigName> flags:
OTHER_LDFLAGS: '-ObjC', ...
```

> Note to self: the format of the CLI command something similar to:
> 
> ```
> pbxproj show 
> pbxproj show <target_name> [(--flags <configuration>) | (--source-files) | (--header-files) | (--resource-files) | (--library-files)]
> ```
> Only one option to display files at the time, otherwise the output can be very unmanageable.

**Comment by kwhat** [comment]
The first two proposed commands would be incredibly helpful working between platforms.  The show target flags would also be quite useful but less critical.  Thanks for this awesome little project.

**Comment by kronenthaler** [comment]
@kwhat i included a rough version of the show command in the 2.5.0 version. The doc is also updated [here](https://github.com/kronenthaler/mod-pbxproj/wiki/cli)

**Comment by kwhat** [comment]
@kronenthaler Thanks for adding this so quickly.  I have been testing and so far everything is pretty awesome.  I did find an issue with some non-source items. Ex:

```
libdyld:
        Product name: libdyld
        Configurations: Debug, Release
        usr|include|mach-o count: 2
        usr|include count: 1
        usr|local|include count: 1
        usr|local|include|mach-o count: 5
        usr|share|man|man1 count: 1
        usr|share|man|man3 count: 7
```
I can seem to get these man pages, includes and other times out of this target.  Both -r and -H return empty. Thanks again.

**Comment by kronenthaler** [comment]
@kwhat it would help me if i can get a sample project.pbxproj. 
I have never seen this kind of build phases on iOS projects, is it a macosx project?

**Comment by kwhat** [comment]
Sure, I am testing with https://opensource.apple.com/tarballs/dyld/dyld-421.2.tar.gz

**Comment by kwhat** [comment]
If you checkout https://opensource.apple.com/tarballs/dyld/dyld-43.tar.gz there are some other issues with that xcode project.  Comments `//` should be stripped?  And it cannot locate the build targets due to some strange structuring.

**Comment by kronenthaler** [comment]
I'm taking a look at the project. The second project is a super old version of the pbxproj format (objectVersion 39, and the current version is 46). That's is probably the reason why it cannot even loaded. Even when i try to open that project on Xcode it complains that the version is not supported.

Regarding the first project. The output is correct, that target is an aggregated target, none of its build phases are sources, headers, resources or frameworks. That explains the lack of output when listing the target alone.
The weird output on the summary is also correct, those are build phases (copy files build phases to be precise), and their names are actually those weird names.

As i see it, there are few options:

1. On the summary list, indicate the type of target.
2. On the detail list, allow to show the copy-build-phases. This is a weird one, because then we should allow the run build phases.
3. On the summary list, filter out the "weird" phases and only allow the ones allowed on the details.
4. On the summary list, display the build phase type (PBXSourcesBuildPhase) instead of its name.
5. On the detail list, put an open parameter for the build phase type to display, instead of -s (or additional to) use --filter PBXSourcesBuildPhase.

@kwhat , I will be waiting for your input as guinea-pig

**Comment by kronenthaler** [comment]
hi @kwhat, both are possible and not conflicting. About your additional requirement to locate the src destinations, that's the thing, they are CopyFilesBuildPhases they point to file references, whatever they may be: docs, images, anything in between. Also, the actual path of the files is possible to obtain but requires some extra work.
In order to list CopyFile build phases we will need also the free form option (5)

I think i have enough to work on it during the weekend.

**Comment by kronenthaler** [comment]
Changes implemented on 2.5.1. I will consider this enhancement as resolved. If anything else arises, please report it.

**Comment by kwhat** [comment]
Thank you so much.
